packages/bridging/src/providers/near-intents/NearIntentsBridgeProvider.ts

```diff
@@ -0,0 +1,370 @@
+    const tokens = adaptTokens(await this.api.getTokens())
+    return {
+      tokens: tokens.filter((token) => token.chainId === params.buyChainId),
+      isRouteAvailable: tokens.length > 0,
+    }
+  }
+
+  async getIntermediateTokens(request: QuoteBridgeRequest): Promise<TokenInfo[]> {
+    if (request.kind !== OrderKind.SELL) {
+      throw new BridgeProviderQuoteError(BridgeQuoteErrors.ONLY_SELL_ORDER_SUPPORTED, { kind: request.kind })
+    }
+
+    const { sellTokenChainId, buyTokenChainId, buyTokenAddress } = request
+
+    const tokens = adaptTokens(await this.api.getTokens())
+    const sourceTokens = tokens.filter((token) => token.chainId === sellTokenChainId)
+    const targetTokens = tokens.filter((token) => token.chainId === buyTokenChainId)
+
+    const wrappedTokenAddresses = Object.values(NEAR_INTENTS_BLOCKCHAIN_TO_NATIVE_WRAPPED_TOKEN_ADDRESS).map((a) =>
+      a.toLowerCase(),
+    )
+    const buyTokens = targetTokens.find((token) => {
+      // Supports both the native token and its wrapped version (e.g. POL and WPOL, where POL is represented by ETH_ADDRESS).
+      if (token.address === ETH_ADDRESS || wrappedTokenAddresses.includes(token.address.toLowerCase()))
+        return token.chainId === buyTokenChainId
+      return token.address.toLowerCase() === buyTokenAddress.toLowerCase()
+    })
+    if (!buyTokens) return []
+
+    // If buyToken is supported, all source tokens can be used to buy buyToken
+    return sourceTokens
+  }
+
+  async getQuote(request: QuoteBridgeRequest): Promise<NearIntentsQuoteResult> {
+    const {
+      sellTokenAddress,
+      sellTokenChainId,
+      buyTokenAddress,
+      buyTokenChainId,
+      account,
+      amount,
+      receiver,
+      validFor,
+      owner,
+    } = request
+    const tokens = await this.api.getTokens()
+
+    const sellToken = getTokenByAddressAndChainId(tokens, sellTokenAddress, sellTokenChainId)
+    const buyToken = getTokenByAddressAndChainId(tokens, buyTokenAddress, buyTokenChainId)
+    if (!sellToken || !buyToken) throw new BridgeProviderQuoteError(BridgeQuoteErrors.NO_ROUTES)
+
+    const { quote, timestamp: isoDate } = await this.api.getQuote({
+      dry: false,
+      swapType: QuoteRequest.swapType.EXACT_INPUT,
+      slippageTolerance: request.slippageBps || 100, // fallback to 1%,
+      originAsset: sellToken.assetId,
+      depositType: QuoteRequest.depositType.ORIGIN_CHAIN,
+      destinationAsset: buyToken.assetId,
+      amount: amount.toString(),
+      refundTo: owner || account,
+      refundType: QuoteRequest.refundType.ORIGIN_CHAIN,
+      recipient: receiver || account,
+      recipientType: QuoteRequest.recipientType.DESTINATION_CHAIN,
+      deadline: calculateDeadline(validFor || 3600),
+    })
+
+    const slippage = (1 - Number(quote.amountOutUsd) / Number(quote.amountInUsd)) * 100
+
+    return {
+      isSell: request.kind === OrderKind.SELL,
+      depositAddress: quote.depositAddress as Hex,
```

Comment:
This is a bit dangerous to just trust the "depositAddress" comming from the API. 

How do we plan to validate this in client side. Near said there will be a way to do it

Reply:
This part won't block this PR, but will block a potential inclusion in CoW Swap as an additional provider. This security risk is very high, and we can't expose users to sending their tokens to an address dictated by an external API